yt_dlp/extractor/vidyard.py

```diff
@@ -0,0 +1,256 @@
+            'url': 'https://play.vidyard.com/iDqTwWGrd36vaLuaCY3nTs.html',
+            'info_dict': {
+                'id': 'lrYeWCqR3UCAzwqUAgegHg',
+                'display_id': '9281009',
+                'ext': 'mp4',
+                'title': 'Lightbox Embed',
+                'thumbnail': 'https://cdn.vidyard.com/thumbnails/spacer.gif',
+                'duration': 39,
+            },
+        },
+    ]
+    _EMBED_REGEX = [
+        # iframe embed
+        r'<iframe[^>]+?src=(["\'])(?P<url>(?:https?:)?//play\.vidyard\.com/[\w-]+.\w+)\1',
+    ]
+    _WEBPAGE_TESTS = [
+        {
+            # URL containing inline/lightbox embedded video
+            'url': 'https://resources.altium.com/p/2-the-extreme-importance-of-pc-board-stack-up',
+            'info_dict': {
+                'id': 'rGGKQEAu8X-APkU68K8U_w',
+                'display_id': '3225198',
+                'ext': 'mp4',
+                'title': 'The Extreme Importance of PC Board Stack Up',
+                'thumbnail': 'https://cdn.vidyard.com/thumbnails/73_Q3_hBexWX7Og1sae6cg/9998fa4faec921439e2c04_small.jpg',
+                'duration': 3422,
+            },
+        },
+    ]
+
+    @classmethod
+    def _extract_embed_urls(cls, url, webpage):
+        # Handle protocol-less embed URLs
+        for embed_url in super()._extract_embed_urls(url, webpage):
+            if embed_url.startswith('//'):
+                embed_url = f'https:{embed_url}'
+            yield embed_url
+
+        # Extract inline/lightbox embeds
+        for embed_elm in re.findall(r'(<img[^>]+class=(["\'])(?:[^>"\']* )?vidyard-player-embed(?: [^>"\']*)?\2[^>]+[^>]*>)', webpage):
+            embed = extract_attributes(embed_elm[0]) or {}
+            uuid = embed.get('data-uuid')
+            if uuid:
+                yield f'https://play.vidyard.com/{uuid}.html'
+
+    def _real_extract(self, url):
+        video_id = self._match_valid_url(url).group('id')
+        video_json = self._fetch_video_json(video_id)
+
+        common_info = {
+            'http_headers': {
+                'referer': 'https://play.vidyard.com/',
+            },
+        }
+
+        if len(video_json['chapters']) == 1:
+            video_info = self._process_video_json(video_json['chapters'][0], video_id, **common_info)
+
+            if video_info['title'] is None or video_info['description'] is None:
+                webpage = self._download_webpage(url, video_id, fatal=False)
+
+                if video_info['title'] is None:
+                    video_info['title'] = self._og_search_title(webpage, default=None) or self._html_extract_title(webpage)
+
+                if video_info['description'] is None:
+                    video_info['description'] = self._og_search_description(webpage, default=None)
+
+            return video_info
+
+        # Playlist
+        return self.playlist_result(
+            [self._process_video_json(chapter, video_id, **common_info) for chapter in video_json['chapters']],
```

Comment:
If applying the suggestion above:
```suggestion
            [self._process_video_json(chapter, video_id) for chapter in video_json['chapters']],
```

Reply:
Referrer handling extracted as suggested as of b8345dd2076859b85857a82e010d4b793978b041